test/basics.jl

```diff
@@ -0,0 +1,56 @@
+test_string_length = 100
+
+#char_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str, UniStr]
+#char_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str]
+char_types = [ASCIIStr, LatinStr, UTF8Str]
+
+##  create type specific test strings
+test_strings_base = Dict{String, Any}()
+for T in char_types
+    test_strings_base["$T"] = join([randchar(Strs.codepoint_type(T)) for i in 1:test_string_length])
```

Comment:
with the above change to `char_types`, you won't need `Strs.codepoint_type(T)`, just `T`